programs/intent-transfer/src/lib.rs

```diff
@@ -28,6 +36,368 @@ pub mod intent_transfer {
+    pub transceiver: UncheckedAccount<'info>,
+
+    /// CHECK: checked in NTT manager program
+    pub emitter: UncheckedAccount<'info>,
+
+    /// CHECK: checked in NTT manager program
+    #[account(mut)]
+    pub wormhole_bridge: UncheckedAccount<'info>,
+
+    /// CHECK: checked in wormhole program
+    #[account(mut)]
+    pub wormhole_fee_collector: UncheckedAccount<'info>,
+
+    /// CHECK: checked in wormhole program
+    #[account(mut)]
+    pub wormhole_sequence: UncheckedAccount<'info>,
+
+    /// CHECK: address is checked, but also verified in NTT manager program
+    #[account(address = WORMHOLE_PROGRAM_ID)]
+    pub wormhole_program: UncheckedAccount<'info>,
+
+    /// CHECK: address is checked
+    #[account(address = cpi::ntt_with_executor::NTT_WITH_EXECUTOR_PROGRAM_ID)]
+    pub ntt_with_executor_program: UncheckedAccount<'info>,
+
+    /// CHECK: address is checked
+    #[account(address = cpi::ntt_with_executor::EXECUTOR_PROGRAM_ID)]
+    pub executor_program: UncheckedAccount<'info>,
+
+    /// CHECK: check not important per https://github.com/wormholelabs-xyz/example-ntt-with-executor-svm/blob/10c51da84ee5deb9dee7b2afa69382ce90984eae/programs/example-ntt-with-executor-svm/src/lib.rs#L74-L76
+    pub ntt_peer: UncheckedAccount<'info>,
+
+    /// CHECK: check not important per https://github.com/wormholelabs-xyz/example-ntt-with-executor-svm/blob/10c51da84ee5deb9dee7b2afa69382ce90984eae/programs/example-ntt-with-executor-svm/src/lib.rs#L78-L80
+    #[account(mut)]
+    pub ntt_outbox_item: Signer<'info>,
+
+    /// CHECK: check not important per https://github.com/wormhole-foundation/native-token-transfers/blob/8bd672c5164c53d5a3f9403dc7ce3450da539450/solana/programs/example-native-token-transfers/src/queue/outbox.rs#L50
+    #[account(mut)]
+    pub ntt_outbox_rate_limit: UncheckedAccount<'info>,
+
+    /// CHECK: checked in NTT manager program
+    #[account(mut)]
+    pub ntt_custody: UncheckedAccount<'info>,
+
+    /// CHECK: checked in NTT with executor program
+    #[account(mut)]
+    pub payee_ntt_with_executor: UncheckedAccount<'info>,
+}
+
+#[derive(Accounts)]
+pub struct BridgeNttTokens<'info> {
+    #[account(seeds = [chain_id::SEED], seeds::program = chain_id::ID, bump)]
+    pub from_chain_id: Account<'info, ChainId>,
+
+    /// CHECK: we check the address of this account
+    #[account(address = instructions::ID)]
+    pub sysvar_instructions: UncheckedAccount<'info>,
+
+    /// CHECK: this is just a signer for token program CPIs
+    #[account(seeds = [INTENT_TRANSFER_SEED], bump)]
+    pub intent_transfer_setter: UncheckedAccount<'info>,
+
+    pub token_program: Program<'info, Token>,
+
+    #[account(mut, token::mint = mint)]
+    pub source: Account<'info, TokenAccount>,
+
+    #[account(
+        init_if_needed,
+        payer = sponsor,
+        seeds = [BRIDGE_NTT_INTERMEDIATE_SEED, source.key().as_ref()],
+        bump,
+        token::mint = mint,
+        token::authority = intent_transfer_setter
```

Comment:
nit: you could set the authority to `ntt_session_authority` right away here and you don't need to approve anymore